pjlib/src/pj/ssl_sock_ossl.c

```diff
@@ -455,12 +455,97 @@ static int openssl_init_count;
 /* OpenSSL application data index */
 static int sslsock_idx;
 
+#if defined(PJ_SSL_SOCK_USE_THREAD_LOCK) && PJ_SSL_SOCK_USE_THREAD_LOCK != 0 \
+    && OPENSSL_VERSION_NUMBER < 0x10100000
```

Comment:
as mentioned above, IMO better check OpenSSL version number check in the config.h.

Reply:
ok

Reply:
version number check shouldn't be on config.h since it needed to include ```opensslv.h``` and other version checking (existing ones) is done on ssl_sock_ossl.c